dev-scripts/dependencies/setup.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+set -euo pipefail
+
+SCRIPT_DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+ROOT_DIR=$(cd "${SCRIPT_DIR}/../.." && pwd)
+BAZEL_DEPS_DIR="$ROOT_DIR/../bazel-deps"
```

Comment:
that'll land the new repo outside of the current tree?
feels awkward, though landing it inside the tree is also awkward. So I don't know :)

Reply:
Managing external repos outside of bazel feels really awkward. I don't have a great solution.

Do you or @kevingessner or @johnynek have a link to the issue or issues preventing us from using `bazel-deps` as a repository rule? It'd be nice to track them. I see [bazel #1673](https://github.com/bazelbuild/bazel/issues/1673) but perhaps there's more to it.

Reply:
#1673 is the only one I know of.  Even if it's in the workspace, it'll be a little bit of wrangling to get it running correctly -- IIRC bazel-deps comes with the `gen_maven_deps.sh` wrapper script because it doesn't work with `bazel run`.  So we'll need to get the various paths right.  Doable but may be a little tricky.